Assets/GhostCollider.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GhostCollider : MonoBehaviour
+{
+    private CharacterController ghostController;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        ghostController = GetComponentInChildren<CharacterController>();
+        ghostController.detectCollisions = false;
+    }
+
+    public void ActivateCollider(bool activate)
+    {
+        Debug.Log("Collider is now active status " + activate);
+        ghostController.detectCollisions = activate;
+    }
+
+    private void OnTriggerEnter(Collider other)
+    {
+        Debug.LogWarning("Collider triggered by ghost");
+        Checkpoint checkPointHit = other.gameObject.GetComponent<Checkpoint>();
```

Comment:
we probably want to use "gameObject.tag == "checkpoint", as getting components is relatively slow.

Reply:
.comparetag and unsure if its faster than getcomponent.

Reply:
Significantly